### DISCUSSION TITLE
Need a neutral volume profile where the phone's volume settting is used

### DISCUSSION THREAD

**Discussion by geoffm** [NightscoutFoundation/xDrip#3233] [Ideas]
Some of us need the app to just use the volume that we chose for our phone alarms instead of forcing loud profiles that arent necessarily always appropriate for the context. This would honor the "prioritize user control" principle.
Otherwise at least provide a "Low" or even "Custom" sound profile because the "Medium" one is too loud.

**Comment by Navid200** [comment]
Have you tried Ascending volume profile?

The request for a low volume profile has existed for years.  We have not granted that because it could lead to serious problems and we cannot blame some things on the user and say that if someone chooses a setting, they are responsible for the outcome.

Please use the ascending volume profile, disable the delay setting and let me know what you think.

https://navid200.github.io/xDrip/docs/Ascending-volume-profile.html

**Comment by geoffm** [comment]
Ascending profile is the worst of them all, it can get teeribly loud. The fact also that it messes with the system volume is really bad because whenever we listen to something, its volune is also raised when the alert goes off. It can instany raise the volume of the music im listening to, it's hurtful and startling. This can be very dangerous when it happens while I'm driving. It's so annoying that im considering just using silent mode instead. We'll see how that goes in preventing bad things happening 
.Providing a non-agressive volume profile could not "lead to serious problems", you should reaxamine your reasoning.You allow a silent profile, wouldn't that lead to serious problems too? Users need to be able to choose for themselves the volume they need. Any low sound will wake me up, I dont want loud sounds to be forced on me by people who believe everyine needs loud sounds like themselves. Loud noises can be stressful and startling for some people. I dont need loud or "medium" volume.

 There's not even a need for messing around with the volumes, just play the sound at the normal volume the phone is set to. Or at least provide the option . If you're so scared of users accidentally chosing a low profile, add a warning popup or something.I will gladly pay for someone to program an extra profile that doesnt modify the volume.Meanwhile, im using sound files that I modified to be more quiet because the high volumes you're forcing on users is sressful, disturbing and potentially dangerous.

 You should really consider that not everyone is like you, some people have different needs and sensibilities. If it has been a recurring request that's because it's a real need out there. Please don't force your personal preferences on everyone. Just provide options. Like that principle that's displayed on the home page.

**Reply by Navid200** [reply]
Please don't reply to an email in order to post here.
It is very hard to read your post.

You can use audio focus to control what happens to the sound of other apps when an alert goes off. https://navid200.github.io/xDrip/docs/AudioFocus.html

I am sorry we are not adding a low volume profile.
You are free to use another app.

**Reply by spanosp** [reply]
@Navid200 @geoffm Silent and Vibrate Only options already exist and arguably these could 'lead to serious problems' equally or more than a Low Volume setting. Could there be a vote for this option? I also beleive it will be a popular choice. Please reconsider, afterall its only an option.

**Reply by Navid200** [reply]
Consider someone who is deaf.  The silent option is for them.  They don't want everyone around them to hear the alert if they themselves cannot hear it.
So, we cannot remove the silent option because as I have stated before, we don't discriminate.  If someone is deaf, they should still be allowed to use xDrip without any impediment forced on them by xDrip.  So, we cannot remove the silent option for that reason.

Some day in the future, I hope to add a mechanism to make vibration mandatory for alerts if the silent option has been chosen.

I think this is a misunderstanding.  If we can see a problem with xDrip, we have no reason not to fix it given time.
No one can help us recreate this problem.

If the problem is that you are listening to music with earphones and an alert raises the volume of the music, I have already told you you can fix it using audio focus.

If you are sleep at night and you don't want the alert to wake everyone else up, I told you to use the ascending volume.  If the phone is near you on the bed stand, you can snooze it as soon as it starts with a 33% volume before it becomes too loud.

But, if you intend to put the phone on the living room table and never touch it, well, yes, it will eventually blast at 100% volume.  But, why would you do that?
Can you explain?

**Reply by spanosp** [reply]
Hi @Navid200

Taking your example, when one is asleep, with the Ascending Volume option, by the time you wake up, find the phone, unlock the phone and silence the alarm, the phone is already at maxmimum volume, waking eveyone around you - Maybe I'm a heavy sleeper but it happens to me all the time :)

Similarly, during the day, you might not have your phone right near you and might not be able to your phone straight away (eg your hands are wet), the alarm will get to maximum volume before you can silence it, disturbing eveyone around you.

If a Low Volume option is a safety concern maybe there could be an Ascending Volume option that goes up to Medium Volume instead or something to that effect.

*Also, for some reason, after changing phones, I can only snooze the alarm if the phone is unlocked and XDrip is open.
If im not mistaken it used to be the case where you could silnce the alarm with the volum keys even if the phone was locked.

TY

**Reply by Navid200** [reply]
That is fair.  Thanks for that.  

Please let me have a conversation with the developers to see if they let me add the option to have ascending go only up to medium.

**Reply by spanosp** [reply]
Thank you for your time

**Reply by Navid200** [reply]
What is the new phone brand and model?

**Reply by spanosp** [reply]
Pixel 8 Pro

**Reply by Navid200** [reply]
I have a Pixel 6 a and I can snooze the alert on the locked screen.
Please have a look at the following page specially what's under the heading "Snooze while the screen is locked".
https://navid200.github.io/xDrip/docs/Snooze.html

Do you have always on display active or disabled?

**Reply by spanosp** [reply]
I had a Pixel 6 and I'm pretty sure I could snooze the alarm while the phone was locked.
I have followed the "Snooze while the screen is locked" but the problem persists.

If "Always on display" = Settings / Display / Lock screen / Always show time and info  then Yes, this is enabled.

**Reply by Navid200** [reply]
What do you see at the following?
Settings &#8722;> Display &#8722;> Lock screen &#8722;> Privacy

**Reply by spanosp** [reply]
= Show all notification content

**Reply by Navid200** [reply]
What do you see in xDrip under the following?
Settings &#8722;> xDrip Display Settings &#8722;> Public Notifications

I have it disabled and it works.  But, I wonder if on Pixel 8, it may need to be enabled.

**Reply by spanosp** [reply]
For me it doesnt work with it enabled or disabled.

**Reply by Navid200** [reply]
Please go to:
Android settings &#8722;> Notifications &#8722;> App notifications

Now, scroll and find xDrip.  There should be an enabled button on the right.  But, you should be able to tap on the left.  Please do.

What do you see?

**Reply by spanosp** [reply]
Show notifications: Enabled
Bubbles: All converdsations can bubble
Allow interuptions: Enabled
Lock screen: Show all notification content
Override Do Not Disturb: Enabled
Allow notification dot: Enabled

**Comment by Navid200** [comment]
@jamorham @tzachi-dar 

Can I add the option to have the ascending volume raise gradually to 70% (medium) instead of to 100% (high)?

To avoid adding yet one more setting to the "Glucose Level Alerts List" page, I suggest replacing the existing "Delay ascending volume" setting with "Ascending profile settings".  This will open a new page that will contain the existing delay ascending volume and new setting that will allow user to choose if the ascending volume raises to 70% or 100%.
Any other setting we may need to add in the future to the ascending volume will go to this page.

My PR will address all of this, and have comments to avoid complicating the code.

**Comment by Navid200** [comment]
There is now a new option for the ascending volume profile.
As of the release of May 15, 2024, you can choose for the ascending volume profile to only ascend to the medium level instead of to the maximum.
